ipn/ipnlocal/local.go

```diff
@@ -352,7 +353,7 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
 
 		b.updateFilter(st.NetMap, prefs)
 		b.e.SetNetworkMap(st.NetMap)
-		if !dnsMapsEqual(st.NetMap, netMap) {
+		if acceptDNS && !dnsMapsEqual(st.NetMap, netMap) {
```

Comment:
I think you'll also need to clear out any existing DNS settings, to account for the case where someone turns off corpdns.

Reply:
Which DNS settings? This path only sets the Magic DNS map of names.

Reply:
Ping @danderson.

Reply:
Ping @danderson again. :)